web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,472 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect } from "react";
+import { FormSelect, FormSelectOption, Radio } from "@patternfly/react-core";
+
+import { _, n_, N_ } from "~/i18n";
+import { deviceSize } from '~/components/storage/utils';
+import { If, Section, SectionSkeleton } from "~/components/core";
+import { noop, useLocalStorage } from "~/utils";
+import { sprintf } from "sprintf-js";
+import { Table, Thead, Tr, Th, Tbody, Td, TreeRowWrapper } from '@patternfly/react-table';
+
+/**
+ * @typedef {import ("~/client/storage").ProposalManager.ProposalSettings} ProposalSettings
+ * @typedef {import ("~/client/storage").ProposalManager.SpaceAction} SpaceAction
+ * @typedef {import ("~/client/storage").DevicesManager.StorageDevice} StorageDevice
+ */
+
+/**
+ * @typedef SpacePolicy
+ * @type {object}
+ * @property {string} name
+ * @property {string} label
+ * @property {string} description
+ */
+
+/** @type {SpacePolicy[]} */
+const SPACE_POLICIES = [
+  {
+    name: "delete",
```

Comment:
NP: I'd use `id` instead of `name`. What do you think?

Reply:
Fine with me.